src/App.js

```diff
@@ -1,5 +1,74 @@
+import { MissionUtils } from "@woowacourse/mission-utils";
+
 class App {
-  async play() {}
+  async play() {
+    let gameOver = false;
+    let correctAnswer;
+
+    MissionUtils.Console.print("숫자 야구 게임을 시작합니다.");
+
+    const answer = [];
+    while (answer.length < 3) {
+      const number = MissionUtils.Random.pickNumberInRange(1, 9);
+      if (!answer.includes(number)) {
+        answer.push(number);
+      }
+    }
+    correctAnswer = answer.join('');
+
+    while (!gameOver) {
+      try {
+        let input = await MissionUtils.Console.readLineAsync("숫자를 입력해주세요: ");
+        input = input.trim();
+
+        if (
+          input.length !== 3 ||
+          input.includes("0") ||
+          input[0] === input[1] ||
+          input[1] === input[2] ||
+          input[0] === input[2]
+        ) {
+          throw new Error("[ERROR] 숫자가 잘못된 형식입니다.");
+        } else {
+          let strike = 0;
+          let ball = 0;
```

Comment:
우테코에서 가이드 해준 것처럼 함수 분리부터 시도해보세요. 예를 들어 윗 라인을 함수로 분리하게되면 결과적으로

```suggestion
    validateInput(input); // 형식이 틀리면 exception 발생

    let strike = 0;
    let ball = 0;
```
위와 같은 진행으로 바뀝니다.

그렇게 분리된 상태의 함수들을 훑어보고, 그룹화 할 수 있다면 
```javascript
class Validator {
    validateBallNumber(input){ ... }

    validateRetryInput(input){ ... }
    
    ...
}

const validator = new Validator();
export { validator };
```

위와 같은 모듈화를 통해 별도의 파일로 분리할 수도 있습니다.

이렇게 분리하게 되면 장점은 읽기도 쉬워지며, 어떤 문제가 생겼을 때 빠르게 고칠 부분을 찾아내어 해당 맥락 안에서 집중할 수 있습니다.

모듈화의 방법은 class 사용말고도 프로토타입, 클로져 등 다양한 방법을 사용할 수 있습니다.

Reply:
확실히 훨씬 간단해져서 너무 보기 좋은 것 같습니다!

함수 분리 -> 그룹화 -> 모듈화

이 순으로 진행해볼 수 있도록 하겠습니다.

정말 감사드립니다! 좋은 하루 되세요~